### PR TITLE
Add security customisation points

### PR DESCRIPTION
**Pull request by jryans**
This adds various customisations point in the app for security related
decisions. By default, these do nothing, but would be customised at the
app level via module replacement (so that no changes are needed here in the
SDK).

See also https://github.com/vector-im/element-web/pull/15475 for usage documentation.

Fixes https://github.com/vector-im/element-web/issues/15350

### CODE REVIEW

File: src/async-components/views/dialogs/security/CreateSecretStorageDialog.js [hunk 1]
```diff
@@ -22,7 +22,7 @@ import {MatrixClientPeg} from '../../../../MatrixClientPeg';
 import FileSaver from 'file-saver';
 import {_t, _td} from '../../../../languageHandler';
 import Modal from '../../../../Modal';
-import { promptForBackupPassphrase } from '../../../../SecurityManager';
+import { getSecretStorageKeyFromAppStorage, promptForBackupPassphrase } from '../../../../SecurityManager';
```
Comment (turt2live) [inline src/async-components/views/dialogs/security/CreateSecretStorageDialog.js:25]: linter is complaining about this

File: src/SecurityManager.ts [hunk 2]
```diff
@@ -352,14 +366,19 @@ export async function accessSecretStorage(func = async () => { }, forceReset = f
                 }
                 console.log("Setting dehydration key");
                 await cli.setDehydrationKey(secretStorageKeys[keyId], dehydrationKeyInfo, "Backup device");
-            } else {
+            } else if (!keyId) {
                 console.log("Not setting dehydration key: no SSSS key found");
+            } else {
+                console.log("Not setting dehydration key: feature disabled");
```
Comment (turt2live) [inline src/SecurityManager.ts:372]: I'm slightly curious if these two should be warnings instead of info

**Review by turt2live — APPROVED** [review]
I think the type failures are a false alarm on this - looks good to me.

File: src/async-components/views/dialogs/security/CreateSecretStorageDialog.js (hunk 1 above)
Comment (jryans) [reply src/async-components/views/dialogs/security/CreateSecretStorageDialog.js:25]: Ah thanks, left overs from a previous attempt, should be fine after removing.

File: src/SecurityManager.ts (hunk 2 above)
Comment (jryans) [reply src/SecurityManager.ts:372]: Ah, I'll upgrade the key not found path to warning at least.

**Pull request merged**